src/test/java/com/google/devtools/build/lib/actions/ActionCacheCheckerTest.java

```diff
@@ -78,7 +86,38 @@ public void setupCache() throws Exception {
     cache = new CorruptibleActionCache(scratch.resolve("/cache/test.dat"), clock);
     cacheChecker =
         new ActionCacheChecker(
-            cache, artifactResolver, new ActionKeyContext(), Predicates.alwaysTrue(), null);
+            cache,
+            artifactResolver,
+            new ActionKeyContext(),
+            action -> true,
+            null,
+            PathFragment.create("bazel-out"));
+    digestHashFunction = DigestHashFunction.SHA256;
+    fileSystem = new InMemoryFileSystem(digestHashFunction);
+    artifactRoot = ArtifactRoot.asDerivedRoot(
+        fileSystem.getPath("/output"),
+        RootType.Output,
+        "bin");
+
+  }
+
+  private byte[] digest(byte[] content) {
+    return digestHashFunction.getHashFunction().hashBytes(content).asBytes();
+  }
+
+  private void enableSaveOutputMetadata() {
```

Comment:
It looks like you should be able to make that into a helper also called from setupCache (with a boolean flag).